test/OpenChat.PlaygroundApp.Tests/Options/GitHubModelsArgumentOptionsTests.cs

```diff
@@ -12,134 +12,173 @@ public class GitHubModelsArgumentOptionsTests
-	}
-
-	[Trait("Category", "UnitTest")]
-	[Theory]
-	[InlineData("--endpoint")]
-	[InlineData("--token")]
-	[InlineData("--model")]
-	public void Given_ArgumentWithoutValue_When_Parse_Invoked_Then_It_Should_Not_Set_Property(string argument)
-	{
-		var config = BuildConfigFrom(("ConnectorType", ConnectorType.GitHubModels.ToString()));
-		var settings = Parse(config, argument);
+    private static IConfiguration BuildConfigWithGitHubModels(
+        string? configEndpoint = Endpoint,
+        string? configToken = Token,
+        string? configModel = Model,
+        string? envEndpoint = null,
+        string? envToken = null,
+        string? envModel = null)
+    {
+        // Base configuration values (lowest priority)
+        var configDict = new Dictionary<string, string?>
+        {
+            ["ConnectorType"] = ConnectorType.GitHubModels.ToString()
+        };
+
+        if (string.IsNullOrWhiteSpace(configEndpoint) == false)
+        {
+            configDict["GitHubModels:Endpoint"] = configEndpoint;
+        }
+        if (string.IsNullOrWhiteSpace(configToken) == false)
+        {
+            configDict["GitHubModels:Token"] = configToken;
+        }
+        if (string.IsNullOrWhiteSpace(configModel) == false)
+        {
+            configDict["GitHubModels:Model"] = configModel;
+        }
+
+        if (string.IsNullOrWhiteSpace(envEndpoint) && string.IsNullOrWhiteSpace(envToken) && string.IsNullOrWhiteSpace(envModel))
+        {
+            return new ConfigurationBuilder()
+                       .AddInMemoryCollection(configDict!)
+                       .Build();
+        }
+
+        // Environment variables (medium priority)
+        var envDict = new Dictionary<string, string?>();
+        if (string.IsNullOrWhiteSpace(envEndpoint) == false)
+        {
+            envDict["GitHubModels:Endpoint"] = envEndpoint;
+        }
+        if (string.IsNullOrWhiteSpace(envToken) == false)
+        {
+            envDict["GitHubModels:Token"] = envToken;
+        }
+        if (string.IsNullOrWhiteSpace(envModel) == false)
+        {
+            envDict["GitHubModels:Model"] = envModel;
+        }
+
+        return new ConfigurationBuilder()
+                   .AddInMemoryCollection(configDict!)  // Base configuration (lowest priority)
+                   .AddInMemoryCollection(envDict!)     // Environment variables (medium priority)
+                   .Build();
+    }
+
+    private static AppSettings Parse(IConfiguration config, params string[] args)
+    {
+        var fullArgs = new List<string>();
+        if (args.Any(a => a.Equals("--connector-type", StringComparison.CurrentCultureIgnoreCase) || a.Equals("-c", StringComparison.CurrentCultureIgnoreCase)) == false)
+        {
+            fullArgs.AddRange(["--connector-type", ConnectorType.GitHubModels.ToString()]);
+        }
+        fullArgs.AddRange(args);
+
+        return ArgumentOptions.Parse(config, [.. fullArgs]);
+    }
```

Comment:
It seems to be redundant. Can we remove it and reflect the change to each test method using it?